libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/useShareDialogBase.ts

```diff
@@ -70,12 +75,37 @@ const useShareDialogState = (
     }, []);
 
     const onGranularGranteeShareChange = useCallback(
-        (grantee: GranteeItem) => {
-            onGranteeAdd(grantee);
+        (grantee: IGranularGranteeUser | IGranularGranteeGroup) => {
+            const originalGrantee = originalGranularGrantees.find((g) => areObjRefsEqual(g.id, grantee.id));
+
+            if (!originalGrantee) {
+                const original = grantees
+                    .filter(isGranularGrantee)
+                    .find((g) => areObjRefsEqual(g.id, grantee.id));
+
+                onGranteeAdd(grantee);
+                setOriginalGranularGrantees((state) => [...state, original]);
+            } else {
+                setGranteesToAdd((state) => {
+                    const granteesToAddWithoutUpdatedGrantee = state.filter(
```

Comment:
I don't think this would be a simplification. In case when the grantee is new and already is an `originalGrantee`, he is not in the `state` so in this case, you would have to add him there at the end of reduce and be sure that he was missing during the loop. The other option would be to remove him also from saved `originalGranularGrantees` every time he is removed from `granteesToAdd`.

Reply:
at least it can be simplified like 
```
const newGranteesToAdd = state.filter(
    (g) => !areObjRefsEqual(g.id, grantee.id),
)
if (!isEqual(originalGrantee.permissions, grantee.permissions)) {
    newGranteesToAdd.push(grantee)
}
return newGranteesToAdd;
```
I'm trying to simplify it even more